ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml

```diff
@@ -70,293 +178,456 @@ StatusDialog {
+                                elide: Text.ElideRight
+                                wrapMode: Text.WordWrap
+                                maximumLineCount: 2
+                            }
+
+                            StatusFlatRoundButton {
+                                Layout.preferredWidth: 32
+                                Layout.preferredHeight: 32
+                                type: StatusFlatRoundButton.Type.Secondary
+                                icon.name: "close"
+                                icon.color: Theme.palette.directColor1
+                                icon.width: 24
+                                icon.height: 24
+                                onClicked: root.communitiesStore.removeFileListItem(model.filePath)
+                            }
+                        }
+
+
+                        StatusBaseText {
+                            Layout.fillWidth: true
+                            text: "%1 %2".arg("⚠").arg(model.errorMessage)
+                            visible: model.errorMessage
+                            font.pixelSize: 13
+                            font.weight: Font.Medium
+                            elide: Text.ElideMiddle
+                            color: Theme.palette.dangerColor1
+                            verticalAlignment: Qt.AlignTop
+                        }
+                    }
                 }
+            }
 
-                validators: [
-                    StatusMinLengthValidator {
-                        minLength: 1
-                        errorMessage: Utils.getErrorMessage(nameInput.errors, qsTr("channel name"))
-                    },
-                    StatusRegularExpressionValidator {
-                        regularExpression: Constants.regularExpressions.alphanumericalExpanded
-                        errorMessage: Constants.errorMessages.alphanumericalExpandedRegExp
+            FileDialog {
+                id: fileDialog
+
+                title: qsTr("Choose files to import")
+                selectMultiple: true
+                nameFilters: [qsTr("JSON files (%1)").arg("*.json")]
+                onAccepted: {
+                    if (fileDialog.fileUrls.length > 0) {
+                        let files = []
+                        for (let i = 0; i < fileDialog.fileUrls.length; i++)
+                            files.push(decodeURI(fileDialog.fileUrls[i].toString()))
+                        root.communitiesStore.setFileListItems(files)
                     }
-                ]
+                }
             }
+        },
 
-            Item {
-                id: spacer1
-                height: 16
-                Layout.fillWidth: true
+        ColumnLayout {
+            id: categoriesAndChannelsView
+            spacing: 24
+
+            readonly property bool canGoNext: root.communitiesStore.discordChannelsModel.hasSelectedItems
+            readonly property var nextAction: function () {
```

Comment:
couldn't it be just:
```suggestion
            function nextAction() {
```
?

Reply:
Nope, it's used by the handlers below...

Reply:
Hmm, does it change anything? Function can be accessed by name in the same as a regular property, right?

Reply:
Hmm, probably it doesn't change anything, it's accessed as:
```qml
const nextAction = currentItem.nextAction
if (typeof(nextAction) == "function") {
    return nextAction()
}
```